db-engine/src/main/java/com/splicemachine/db/impl/sql/conn/GenericLanguageConnectionContext.java

```diff
@@ -398,6 +401,15 @@ public GenericLanguageConnectionContext(
         String logQueryPlanProperty=PropertyUtil.getCachedDatabaseProperty(this,"derby.language.logQueryPlan");
         logQueryPlan=Boolean.valueOf(logQueryPlanProperty);
 
+        try {
+            String valueString = (String) getTransactionCompile().getProperty("derby.language.tableLimitForExhaustiveSearch");
```

Comment:
Please use PropertyUtil.getCachedDatabaseProperty() instead to avoid a lookup into the Property conglomerate always.

Reply:
Done.